synapse_net/training/domain_adaptation.py

```diff
@@ -109,24 +294,29 @@ def mean_teacher_adaptation(
         if os.path.isdir(source_checkpoint):
             model = torch_em.util.load_model(source_checkpoint)
         else:
-            model = torch.load(source_checkpoint, weights_only=False)
+            model = torch.load(source_checkpoint)
         reinit_teacher = False
 
     optimizer = torch.optim.Adam(model.parameters(), lr=1e-4)
     scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, mode="min", factor=0.5, patience=5)
 
     # self training functionality
-    pseudo_labeler = self_training.DefaultPseudoLabeler(confidence_threshold=confidence_threshold)
+    if train_background_mask_paths is not None:
```

Comment:
You now also have to select the correct trainer class here. In the first case (i.e. `train_background_mask_paths is not None`), you have to define `trainer_class = MeanTeacherTrainerWithBackgroundMask`